client/js/transports/websocket.mdx

```diff
@@ -0,0 +1,210 @@
+  recorderSampleRate?: number;
+  playerSampleRate?: number;
+};
+```
+
+#### Properties
+
+<ParamField name="ws_url" type="string">
+  URL of the WebSocket server. This is the endpoint your client will connect to for WebSocket communication.
+</ParamField>
+
+<ParamField name="serializer" type="WebSocketSerializer" default="ProtobufFrameSerializer">
+  The serializer to use for encoding/decoding messages sent over the WebSocket connection. The websocket-transport package provides two serializer options:
+  - `ProtobufFrameSerializer`: Uses Protocol Buffers for serialization.
+  - `TwilioSerializer`: Uses Twilio's serialization format.
+</ParamField>
+
+<ParamField name="recorderSampleRate" type="number">
+  Sample rate for which to encode the audio input. Default is `16000`.
+</ParamField>
+
+<ParamField name="playerSampleRate" type="number">
+  Sample rate for which to decode the incoming audio for output. Default is `24000`.
+</ParamField>
+
+### TransportConnectionParams
+
+The `WebSocketTransport` takes the same options as the constructor; `WebSocketTransportOptions`. Anything provided here will override the defaults set in the constructor. The `ws_url` is required to establish a connection.
+
+<CodeGroup>
+```typescript client
+pcClient.connect({
+  ws_url: 'http://localhost:8765/ws'
+});
+// OR...
+pcClient.connect({
+  endpoint: 'https://your-server/connect', // returns { ws_url }
+});
+```
+
+```python server
+# See
+# https://github.com/pipecat-ai/pipecat/blob/main/examples/websocket/server/server.py
+# for a complete example of how to implement the server-side endpoint.
+@app.websocket("/ws")
+async def websocket_endpoint(websocket: WebSocket):
+    await websocket.accept()
+    print("WebSocket connection accepted")
+    try:
+        await run_bot(websocket)
+    except Exception as e:
+        print(f"Exception in run_bot: {e}")
+
+
+@app.post("/connect")
+async def bot_connect(request: Request) -> Dict[Any, Any]:
+    server_mode = os.getenv("WEBSOCKET_SERVER", "fast_api")
+    if server_mode == "websocket_server":
+        ws_url = "ws://localhost:8765"
+    else:
+        ws_url = "ws://localhost:7860/ws"
+    return {"ws_url": ws_url}
+```
+
+```python bot
+# See
+# https://github.com/pipecat-ai/pipecat/blob/main/examples/websocket/server/bot_websocket_server.py
+# for a complete example of a bot script using the WebSocketTransport.
+from pipecat.serializers.protobuf import ProtobufFrameSerializer
+from pipecat.transports.network.websocket_server import (
+    WebsocketServerParams,
+    WebsocketServerTransport,
+)
+
+async def run_bot_websocket_server():
+    ws_transport = WebsocketServerTransport(
+        params=WebsocketServerParams(
+            serializer=ProtobufFrameSerializer(),
+            audio_in_enabled=True,
+            audio_out_enabled=True,
+            add_wav_header=False,
+            vad_analyzer=SileroVADAnalyzer(),
+            session_timeout=60 * 3,  # 3 minutes
+        )
+    )
```

Comment:
I guess the most common use case would be to use with the Fast API websocket, so maybe replace for this:

```
async def run_bot(websocket_client):
    ws_transport = FastAPIWebsocketTransport(
        websocket=websocket_client,
        params=FastAPIWebsocketParams(
            audio_in_enabled=True,
            audio_out_enabled=True,
            add_wav_header=False,
            vad_analyzer=SileroVADAnalyzer(),
            serializer=ProtobufFrameSerializer(),
        ),
    )
```

Reply:
done